### PR TITLE
feat: add --timeout flag to curate and query commands

### PR DESCRIPTION
**Pull request by andreivladmatei**
## What

Add a `--timeout` flag (in seconds, default: 300) to both `curate` and `query` commands, allowing users to configure the task completion timeout.

Also fixes the hardcoded "Task timed out after 5 minutes" error message in `task-client.ts` to dynamically reflect the configured timeout value.

## Why

Local models are slower than cloud providers and frequently exceed the hardcoded 5-minute timeout, causing curate and query tasks to fail before completion. This change allows users to increase the timeout as needed (e.g., `brv curate "context" --timeout 600` for 10 minutes).

## Changes

- **`src/oclif/commands/curate/index.ts`** — Add `--timeout` flag (integer, default 300, min 10), pass `timeoutMs` to `waitForTaskCompletion`
- **`src/oclif/commands/query.ts`** — Same `--timeout` flag and passthrough
- **`src/oclif/lib/task-client.ts`** — Make timeout error message dynamic instead of hardcoded "5 minutes"
- **`test/commands/curate.test.ts`** — Add tests for timeout flag acceptance and default behavior

## Backward compatible

Default timeout remains 300s (5 minutes). No behavior change unless `--timeout` is explicitly provided.

## Testing

- TDD approach: tests written first, verified they fail, then implemented
- All 5228 tests pass
- `npm run lint` — 0 errors
- `npm run typecheck` — clean
- `npm run build` — clean

### CODE REVIEW

File: src/oclif/commands/curate/index.ts [hunk 1]
```diff
@@ -83,6 +87,11 @@ Bad examples:
       description: 'Output format (text or json)',
       options: ['text', 'json'],
     }),
+    timeout: Flags.integer({
+      default: 300,
+      description: 'Task completion timeout in seconds (default: 300)',
```
Comment (bao-byterover) [inline src/oclif/commands/curate/index.ts]: > **Must fix**: Duplicate default in `--help` output

Line 92:
```typescript
description: 'Task completion timeout in seconds (default: 300)',
```

I ran `brv curate --help` and saw:
```
--timeout=<value>  [default: 300] Task completion timeout in seconds (default: 300)
```
oclif adds `[default: 300]` automatically from line 91. Remove `(default: 300)` from the description:
```typescript
description: 'Task completion timeout in seconds',
```

Same fix needed in `src/oclif/commands/query.ts:56`.

File: src/oclif/commands/curate/index.ts [hunk 2]
```diff
@@ -83,6 +87,11 @@ Bad examples:
...
+      min: 10,
```
Comment (bao-byterover) [inline src/oclif/commands/curate/index.ts]: > **Must fix**: Add `max` to prevent `setTimeout` overflow

```typescript
timeout: Flags.integer({
  default: 300,
  description: 'Task completion timeout in seconds (default: 300)',
  min: 10,
}),
```

Node.js `setTimeout` overflows at `2^31-1` ms (~24.8 days). Without `max`, a very large value like `--timeout 3000000` would cause `setTimeout` to fire immediately instead of waiting — the opposite of what the user expects.

```typescript
timeout: Flags.integer({
  default: 300,
  description: 'Task completion timeout in seconds',
  max: 3600,
  min: 10,
}),
```

Same fix needed in `src/oclif/commands/query.ts:54-58`.

File: test/commands/curate.test.ts [hunk 3]
```diff
@@ -459,4 +459,41 @@ describe('Curate Command', () => {
       expect(json.data).to.not.have.property('pendingReview')
     })
   })
+
+  // ==================== Timeout Flag ====================
+
+  describe('timeout flag', () => {
```
Comment (bao-byterover) [inline test/commands/curate.test.ts:523]: > **Must fix**: Query command has no timeout tests

Tests are added here for curate's `--timeout`, but `test/commands/query.test.ts` (17 tests) has zero coverage for `--timeout`. Our repo requires test coverage for all critical paths (CLAUDE.md: "80% coverage minimum, critical paths must be covered").

`TestableQueryCommand` and the mock patterns already exist in `test/commands/query.test.ts` — you can mirror these curate tests there.

File: src/oclif/commands/curate/index.ts [hunk 4]
```diff
@@ -364,6 +374,7 @@ Bad examples:
             }
           },
           taskId,
+          timeoutMs: (flags.timeout ?? 300) * 1000,
```
Comment (bao-byterover) [inline src/oclif/commands/curate/index.ts]: > **Suggestion** (non-blocking)

```typescript
timeoutMs: (flags.timeout ?? 300) * 1000,
```

This line is inside the `else` branch (non-detach mode, line 313). But when `--detach` is used, the `if` branch at line 299 returns immediately without ever reaching `waitForTaskCompletion`. This means `--timeout` is silently ignored with `--detach` — the user gets no feedback that their flag had no effect.

Consider adding a note in the detach branch (around line 299):
```typescript
if (flags.detach) {
  if (flags.timeout !== 300) {
    this.log('Note: --timeout has no effect with --detach')
  }
  // ... existing detach logic
}
```

File: src/oclif/lib/task-client.ts [hunk 5]
```diff
@@ -155,12 +155,13 @@ export function waitForTaskCompletion(options: WaitForTaskOptions, log: (msg: st
       if (!completed) {
         completed = true
         cleanup()
+        const timeoutMessage = `Task timed out after ${timeoutMs / 1000}s`
```
Comment (bao-byterover) [inline src/oclif/lib/task-client.ts:164]: > **Suggestion** (non-blocking)

```typescript
const timeoutMessage = `Task timed out after ${timeoutMs / 1000}s`
```

This works correctly. One thought: "Task completion timeout" in the flag description could imply the task itself gets killed after this duration. In reality, only the CLI stops waiting — the daemon task continues running. If you want to clarify, consider updating the flag description to:
```typescript
description: 'Maximum seconds to wait for task completion',
```

**Review by bao-byterover — COMMENTED** [review]
Thanks for the PR! This solves a real problem for local model users. I checked out the branch, ran all tests, typecheck, lint, and verified the `--help` output.

**Results**: typecheck clean, lint 0 errors, curate tests 24 passing, query tests 17 passing.

3 items to fix before merge, plus 2 optional suggestions. Details are in inline comments below.

### What I liked

The approach of surfacing the existing `timeoutMs` parameter in `WaitForTaskOptions` rather than adding new infrastructure is clean and minimal. Input validation (`min: 10`) and the dynamic error message fix are nice touches.

File: src/oclif/commands/curate/index.ts [hunk 6]
```diff
@@ -297,6 +308,10 @@ Bad examples:
     }
 
     if (flags.detach) {
+      if (flags.timeout !== 300) {
```
Comment (bao-byterover) [inline src/oclif/commands/curate/index.ts]: This one's on me — my original suggestion to add the warning didn't account for JSON mode. Sorry for the back-and-forth.

The issue: `this.log()` writes to stdout, so in JSON mode the output becomes:
```
Note: --timeout has no effect with --detach
{"command":"curate","data":{"status":"queued",...},"success":true}
```

I verified by running `brv curate "test" --detach --timeout 600 --format json` — the plain text line before the JSON breaks `jq` / `JSON.parse()` piping.

Simple fix — add a format check so the warning only prints in text mode:

```typescript
if (flags.timeout !== 300 && format !== 'json') {
  this.log('Note: --timeout has no effect with --detach')
}
```

That's it. The existing tests should still pass since they use text mode. Thanks for your patience — the warning feature itself is great, just this one edge case.

**Review by bao-byterover — COMMENTED** [review]
Thanks for the quick turnaround! All 3 must-fix items are addressed and both suggestions are implemented. Typecheck, lint, and all tests pass.

One new issue introduced by the detach warning - details in inline comment. Fix is a one-liner, then this is good to merge.

**Verification results:**
| Check | Result |
|-------|--------|
| `npm run typecheck` | Clean |
| `npm run lint` | 0 errors |
| Curate tests | 26 passing (+2 detach warning tests) |
| Query tests | 20 passing (+3 timeout tests) |
| `brv curate --help` | `[default: 300] Maximum seconds to wait for task completion` - clean |
| `brv curate "test" --detach --timeout 600 --format json` | **Bug** - see inline comment |

**Review by bao-byterover — COMMENTED** [review]
Looks good. Before merging, please name the magic numbers in the `--timeout` flag definition. Right now `300`, `3600`, and `10` are literals in both `curate/index.ts` and `query.ts`, and `300` also shows up in the warning check and the `?? 300` fallback. Naming them once removes the drift risk if the defaults ever change.

Each command keeps its own `Flags.integer({...})` block — that part is fine, don't share it. Only the literals need to become named constants.

### `src/oclif/lib/task-client.ts`

Add three exported constants next to the existing `DEFAULT_TIMEOUT_MS`, and derive `DEFAULT_TIMEOUT_MS` from the seconds constant so the `5 * 60 * 1000` literal goes away too:

```ts
/** Default timeout for task completion (seconds) — shared across curate/query commands. */
export const DEFAULT_TIMEOUT_SECONDS = 300
/** Minimum --timeout value accepted from the CLI (seconds). */
export const MIN_TIMEOUT_SECONDS = 10
/** Maximum --timeout value accepted from the CLI (seconds). */
export const MAX_TIMEOUT_SECONDS = 3600
/** Default timeout for task completion (ms) */
const DEFAULT_TIMEOUT_MS = DEFAULT_TIMEOUT_SECONDS * 1000
```

### `src/oclif/commands/curate/index.ts`

Extend the existing `task-client.js` import to include `DEFAULT_TIMEOUT_SECONDS`, `MAX_TIMEOUT_SECONDS`, and `MIN_TIMEOUT_SECONDS`. Then replace the three literals in the `static flags` block:

```ts
timeout: Flags.integer({
  default: DEFAULT_TIMEOUT_SECONDS,
  description: 'Maximum seconds to wait for task completion',
  max: MAX_TIMEOUT_SECONDS,
  min: MIN_TIMEOUT_SECONDS,
}),
```

Replace the two remaining `300` literals in the same file:

- `if (flags.timeout !== 300 && format !== 'json')` → `if (flags.timeout !== DEFAULT_TIMEOUT_SECONDS && format !== 'json')`
- `timeoutMs: (flags.timeout ?? 300) * 1000` → `timeoutMs: (flags.timeout ?? DEFAULT_TIMEOUT_SECONDS) * 1000`

No changes to `CurateFlags`, no changes to the `rawFlags` remap, just the four literal replacements.

### `src/oclif/commands/query.ts`

Same pattern. Extend the `task-client.js` import to include the three constants. Replace the literals in the `static flags` block exactly as above. Replace the remaining `?? 300` fallback:

```ts
// before
await this.submitTask({client, format, projectRoot, query: args.query, timeoutMs: (flags.timeout ?? 300) * 1000})

// after
await this.submitTask({
  client,
  format,
  projectRoot,
  query: args.query,
  timeoutMs: (flags.timeout ?? DEFAULT_TIMEOUT_SECONDS) * 1000,
})
```

### Tests

No test changes required. I verified: no test in `test/commands/curate.test.ts` or `test/commands/query.test.ts` hardcodes `300` for this flag. The tests either pass `--timeout 600` explicitly or call the command without `--timeout` at all.

### Verification

After the edits, please run:

```bash
npm run lint && npm run typecheck && npx mocha --forbid-only "test/commands/curate.test.ts" "test/commands/query.test.ts"
```

Help output stays identical, oclif evaluates the constants at flag registration, so users still see `[default: 300] Maximum seconds to wait for task completion`.

**Comment by bao-byterover** [comment]
Hi @andreivladmatei — thanks for the PR! 🙏

  This branch now has a merge conflict with main that needs to be resolved before we can merge. Only one file is affected:

  - src/oclif/commands/query.ts

  To resolve, please update your branch against latest main. Either approach works:

  After the conflict is resolved, please make sure npm run build, npm run lint, npm run typecheck, and npm test still pass locally. Let me know if you run into any trouble — happy to
  help!

**Review by bao-byterover — APPROVED** [review]
Thanks a lot for your contribution, we really appreciate it.
We're planning to include this in the next release.
